src/govuk/i18n.unit.test.mjs

```diff
@@ -151,13 +151,37 @@ describe('I18n', () => {
   })
 
   describe('pluralisation', () => {
-    it('throws an error if a required plural form is not provided ', () => {
+    it('falls back to `other` if a plural form is not provided', () => {
       const i18n = new I18n({
         'test.other': 'testing testing'
       }, {
         locale: 'en'
       })
-      expect(() => { i18n.t('test', { count: 1 }) }).toThrowError('i18n: Plural form ".one" is required for "en" locale')
+      expect(i18n.t('test', { count: 1 })).toBe('testing testing')
+    })
+
+    it('logs a console warning when falling back to `other`', () => {
```

Comment:
I'm seeing `.many`, `.few` and `.two` logged (not just `.one`)

Shall we do one of those fancy `it.each([])` and make sure each plural form has a test?

Looks good

Reply:
This came up on Slack, so bringing over the points here

1. Do we suppress `.two`, `.few`, `.many` warnings, for all locales?
2. Do we suppress `.two`, `.few`, `.many` warnings, for English only?
3. Treating all locales equally, is it correct to need duplicate strings by design?

```js
var TRANSLATIONS_DEFAULT = {
  // Characters
  charactersUnderLimit: {
    one: 'You have %{count} character remaining',
    two: 'You have %{count} characters remaining',
    few: 'You have %{count} characters remaining',
    many: 'You have %{count} characters remaining',
    other: 'You have %{count} characters remaining'
  },
  charactersAtLimit: 'You have 0 characters remaining',
  charactersOverLimit: {
    one: 'You have %{count} character too many',
    two: 'You have %{count} characters too many',
    few: 'You have %{count} characters too many',
    many: 'You have %{count} characters too many',
    other: 'You have %{count} characters too many'
  },
  // Words
  wordsUnderLimit: {
    one: 'You have %{count} word remaining',
    two: 'You have %{count} words remaining',
    few: 'You have %{count} words remaining',
    many: 'You have %{count} words remaining',
    other: 'You have %{count} words remaining'
  },
  wordsAtLimit: 'You have 0 words remaining',
  wordsOverLimit: {
    one: 'You have %{count} word too many',
    two: 'You have %{count} words too many',
    few: 'You have %{count} words too many',
    many: 'You have %{count} words too many',
    other: 'You have %{count} words too many'
  }
}
```

>To silence the warning, they'd need to pass test.one even though it works exactly the same as other. Should we be trying to enforce passing all plural forms in code when they might not actually be needed?

👆 Above would be the Character Count defaults if we needed all strings

Reply:
Different plural forms are required depending on which locale the Character Count is in.

When in the default 'en' locale, only 'one' and 'other' are required.

So, I think the answers are:

1. We should warn when a plural form is missing that we try to use based on the locale, so 'it depends on the locale'
2. There shouldn't be any cases where we warn for a character count in the `en` locale, because we have all of the plural forms we need to localise in English ('one' and 'other')
3. I think that's what I'm asking with my question about the sheep. But I don't think we'd add plural forms for anything other than 'one' and 'other' to the defaults as the defaults are only designed to be used in an `en` locale – we'd expect if the locale is something other than `en` that we'd have a different set of translations passed with the correct plural forms for the locale. This PR is trying to address the edge case where that doesn't happen.

Reply:
@36degrees Perfect, thanks for the answers

>Shall we do one of those fancy it.each([]) and make sure each plural form has a test?

Do you want to add tests to confirm?

1. English: Warnings **do not** get logged for missing `.two`, `.few`, `.many`
2. Others: Warnings **do** get logged for missing `.two`, `.few`, `.many`

We've got this one already, but won't harm to document this decision about the others in test form

```js
expect(consoleWarn).toHaveBeenCalledWith(
  'i18n: Missing plural form ".one" for "en" locale. Falling back to ".other".'
)
```

Reply:
Yeah, makes sense

I like to think if a less experienced developer touches these lines in future, and they break something, the tests should be there to explain exactly why the change broke something—gives them a safety net.

Also, if we fix a Welsh bug, add a Welsh test case 😆

If it feels better in the Puppeteer tests, further away from I18n, can do that?

Reply:
I've added a failing test to the Character Count component in c757025267f9df0f4d6dfcfb86f1943c7a30afb9.